mss/L2Cache.cpp

```diff
@@ -0,0 +1,657 @@
+
+                ILOG("Request found in miss_pending_buffer_ with SrcUnit : " << (*req)->getSrcUnit());
+
+                (*req)->setDestUnit((*req)->getSrcUnit());
+                (*req)->setSrcUnit(L2ArchUnit::BIU);
+                (*req)->setCacheState(L2CacheState::RELOAD);
+
+                if (pipeline_req_queue_.numFree() > 0) {
+                    pipeline_req_queue_.push(*req);
+                }
+                else {
+                    sparta_assert("pipeline_req_queue_ is full. Check the sizing.")
+                }   
+                
+                // Check if this was the last occuring
+                auto iter = req; ++iter;
+                auto next_req = std::find_if(iter, miss_pending_buffer_.end(), is_addr_present);
+                
+                if (next_req == miss_pending_buffer_.end()) {
+                    
+                    // When the find_if() returns .end(), free the entry in the biu_resp_queue_
+                    biu_resp_queue_.erase(biu_resp_queue_.begin());
+                }
+
+                // Free the entry in the miss_pending_buffer_
+                miss_pending_buffer_.erase(req);
+            }
+        }
+        else if (arbitration_winner == Channel::ICACHE) {
+            
+            const auto &reqPtr = std::make_shared<olympia::MemoryAccessInfo>(icache_req_queue_.front());
+            
+            reqPtr->setSrcUnit(L2ArchUnit::ICACHE);
+            reqPtr->setDestUnit(L2ArchUnit::ICACHE);
+
+            pipeline_req_queue_.push(reqPtr);
+            ILOG("ICache request is sent to Pipeline_req_Q!");
+
+            icache_req_queue_.erase(icache_req_queue_.begin());
+            
+            // Send out the ack to ICache for credit management
+            ev_handle_l2cache_icache_ack_.schedule(sparta::Clock::Cycle(0));
+        }
+        else if (arbitration_winner == Channel::DCACHE) {
+            
+            const auto &reqPtr = std::make_shared<olympia::MemoryAccessInfo>(dcache_req_queue_.front());
+            
+            reqPtr->setSrcUnit(L2ArchUnit::DCACHE);
+            reqPtr->setDestUnit(L2ArchUnit::DCACHE);
+
+            pipeline_req_queue_.push(reqPtr);
+            ILOG("DCache request is sent to Pipeline_req_Q!");
+
+            dcache_req_queue_.erase(dcache_req_queue_.begin());
+            
+            // Send out the ack to DCache for credit management
+            ev_handle_l2cache_dcache_ack_.schedule(sparta::Clock::Cycle(0));
+        }
+
+        // Try to issue a request to l2cache_pipeline_
+        ev_issue_req_.schedule(sparta::Clock::Cycle(0));
+
+        // Schedule a ev_create_req_ event again to see if the the new request 
+        // from any of the requestors can be put into pipeline_req_queue_
+        if (   !biu_resp_queue_.empty()
+            || !icache_req_queue_.empty()
+            || !dcache_req_queue_.empty() )
+            
+            ev_create_req_.schedule(sparta::Clock::Cycle(1));
```

Comment:
Our coding style guide rule 17 requires that all conditional statements, even if just one line, must be in `{}`.  Reason for this:  I personally lost _years_ (I exaggerate 😆 ) of time debugging code like this:
```
if (condition) x = 3;
    y = 4;
```
WHY IS y always getting 4??!?  Oh.  Missing `{}`.